server/index.js

```diff
@@ -152,29 +166,24 @@ io.on("connection", (socket) => {
       }
     });
 
-    socket.on("dropSelection", ({ name }) => {
-      const player = players.get(name);
-
-      if (player) {
-        player.resetSelected();
-      }
-    });
-
     socket.on("action:start", ({ name }) => {
       const player = players.get(name);
 
-      const item = getItem(player.selectedPlayer);
-
       if (!player.canPerformAction()) {
         // @TODO: send message that action can't be performed #164
         return;
       }
 
-      if (item && player.addToBackpack(item)) {
-        player.energyUse(player.action);
+      const durationTicks = getDuration(player.selectedPlayer);
 
-        io.to(player.socketId).emit("player:backpack", player.backpack);
-      }
+      player.actionDurationTicks = 0;
+      player.actionDurationMaxTicks = durationTicks;
+
+      player.energyUse(player.action);
+      io.to(player.socketId).emit(
+        "action:progress",
+        Math.floor(durationTicks * FRAME_IN_MS)
+      );
```

Comment:
Is this correct? First `player.actionDurationMaxTicks = durationTicks;` is assigned and then `        Math.floor(durationTicks * FRAME_IN_MS)` reported as a progress?

Reply:
yes, it's correct
duractionTicks is in ticks... it's not miliseconds or seconds, it is just a number.
By the multiplication `durationTicks * FRAME_IN_MS` we get miliseconds that are required by tweens later on the client side.
Our current FRAME_IN_MS is 1000/30 = 33,(3).
Eg. 15 * 33,3 = 499,5 ms, which is 5s, which is good :)

Reply:
Actually I was asking about the progress itself, but now I can see that this is actually not reporting progress but starting counter on client side instead:
```
  game.socket.on("action:progress", (duration) => {
    game.mainPlayer.actionStart(duration);
  });
```
Please rename it to `action:start` or `action:progressStart`

Reply:
Yes, it's just the value for telling the progress bar how long the animation should take.
I renamed it :)